### PR TITLE
Fixed SDRF generation bugs.

### PR DESCRIPTION
**Pull request by sandeepreddy792**
Version upgrades:
Upgraded GWT to 2.9.0
Limpopo upgrade to 2.0.1
Guava upgrade to 21.0
asm upgrade to 9.2 (Internal dependency for GWT compilation)

### CODE REVIEW

File: app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java [hunk 1]
```diff
@@ -518,17 +561,16 @@ public Integer apply(Protocol protocol) {
...
-                    protocolNode = createNode(ProtocolApplicationNode.class, nodeName);
-                    protocolNode.protocol = protocol.getName();
-                    if (protocol.hasPerformer()) {
-                        PerformerAttribute attr = new PerformerAttribute();
-                        attr.setAttributeValue(protocol.getPerformer());
-                        protocolNode.performer = attr;
-                    }
+            ProtocolApplicationNode protocolNode = null;
+            if (protocol.isAssigned()) {
+                protocolNode = createNode(ProtocolApplicationNode.class, nodeName);
```
Comment (alfonsomunozpomer) [inline app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java:566]: Can `createNode` return `null`?

File: app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java [hunk 2]
```diff
@@ -279,10 +325,7 @@ private String nodeId(Class<?> clazz, String name) {
         if (isNullOrEmpty(nodeName)) {
             return createFakeNode(clazz);
         }
-        T node = getNode(clazz, nodeName);
-        if (null == node) {
-            node = createNode(clazz, nodeName);
-        }
+        T node = createNode(clazz, nodeName);
```
Comment (alfonsomunozpomer) [inline app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java:328]: :+1:
Compare this to my comment about `protocolNode` in the `connect` method.

**Review by alfonsomunozpomer — COMMENTED** [review]
This looks very good! Just a minor question regarding code clarity before approval.

File: app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java (hunk 1 above)
Comment (sandeepreddy792) [reply app/core/src/main/java/uk/ac/ebi/fg/annotare2/core/magetab/MageTabGenerator.java]: Yes, `createNode` will never return `null`. I have simplified this block by removing unnecessary null checks.
But here we need additional variable `protocolNode` to hold current node and link it to 'prev' which is source node. And then `prev' has been changed to current node.

**Review by alfonsomunozpomer — APPROVED** [review]
All clean now. Ready for lift-off! :rocket:

**Pull request merged**